ui.frontend/src/utils.js

```diff
@@ -206,4 +206,11 @@ export default class Utils {
         }
         return prefillJson;
     }
+
+    /**
+     * detects safari browser
+     */
+    static isSafari() {
+        return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
```

Comment:
we are trying to make ui.frontend  module to be as isomorphic as possible.
pls don't assume that window.navigator would be available in the client.
additional null check for navigator is required.
method name can be `isUserAgent(agentName)`

Reply:
Done.